src/fair/simulation.py

```diff
@@ -67,27 +69,35 @@ def __init__(
             self.preferred_courses += topic
 
         self.total_courses = rng.integers(lower_max_courses, upper_max_courses + 1)
-        all_courses = [item.value(course) for item in schedule]
+        all_courses = [(item.value(course), item.value(section)) for item in schedule]
         self.all_courses_constraint = PreferenceConstraint.from_item_lists(
             schedule,
             [all_courses],
             [self.total_courses],
-            course,
+            [course, section],
             sparse,
         )
-        undesirable_courses = list(set(all_courses).difference(self.preferred_courses))
+        undesirable_courses = [
+            (item.value(course), item.value(section))
+            for item in schedule
+            if item not in self.preferred_courses
+        ]
         self.undesirable_courses_constraint = PreferenceConstraint.from_item_lists(
             schedule,
             [undesirable_courses],
             [0],
-            course,
+            [course, section],
             sparse,
         )
+        topic_values = [
+            [(item.value(course), item.value(section)) for item in topic]
+            for topic in self.preferred_topics
+        ]
         self.topic_constraint = PreferenceConstraint.from_item_lists(
```

Comment:
When creating students from responses in the course allocation data repo, there is no `topic_constraint`, just `preferred_courses_constraint`. In this case, there is no `topic_values`, instead, we have  `preferred_courses`. Which means that, instead of having a list of lists of ScheduleItems, we just have a list of ScheduleItems. 

Currently the code is the following for the survey student:

```
self.preferred_courses_constraint = PreferenceConstraint.from_item_lists(
    schedule,
    [self.preferred_courses],
    [self.total_courses],
    [course, section],
    sparse,
) 
```

Would it be enough to do `[[self.preferred_courses]]` instead of just `[self.preferred_courses]` to make it work?